[cpp/RE2] USE_RE2=ON fails to switch on RE2 regexp

**Build Environment**: gcc 4.8.1, RE2 installed from source
**Steps**:from libphonenumber/cpp
$ mkdir build
$ cd build
$ cmake -D USE_ICU_REGEXP=OFF -D USE_RE2=ON ..
$ make
**Output**: snipped successful output
...
Linking CXX executable libphonenumber_test
CMakeFiles/libphonenumber_test.dir/test/phonenumbers/regexp_adapter_test.cc.o: In function `i18n::phonenumbers::ICURegExpFactory::ICURegExpFactory()':
regexp_adapter_test.cc:(.text._ZN4i18n12phonenumbers16ICURegExpFactoryC2Ev[_ZN4i18n12phonenumbers16ICURegExpFactoryC5Ev]+0x1f): undefined reference to`vtable for i18n::phonenumbers::ICURegExpFactory'
collect2: error: ld returned 1 exit status
make[2]: **\* [libphonenumber_test] Error 1
make[1]: **\* [CMakeFiles/libphonenumber_test.dir/all] Error 2
make: **\* [all] Error 2
$
**Actual**: ICU is being switched on, RE2 is not.
**Expected**: RE2 should be switched on, not ICU.
**Fix**: In file cpp/test/phonenumbers/regexp_adapter_test.cc,
:%s/USE_RE2/I18N_PHONENUMBERS_USE_RE2
**Post-fix**: Output as Expected.

--- a/cpp/src/phonenumbers/phonenumbermatcher.cc
+++ b/cpp/src/phonenumbers/phonenumbermatcher.cc
@@ -53,7 +53,7 @@
 
 #ifdef I18N_PHONENUMBERS_USE_RE2
 #include "phonenumbers/regexp_adapter_re2.h"
-#endif  // I18N_PHONENUMBERS_USE_RE2_AND_ICU
+#endif  // I18N_PHONENUMBERS_USE_RE2
 
 using std::map;
 using std::numeric_limits;

--- a/cpp/test/phonenumbers/regexp_adapter_test.cc
+++ b/cpp/test/phonenumbers/regexp_adapter_test.cc
@@ -26,11 +26,11 @@
 #include "phonenumbers/stl_util.h"
 #include "phonenumbers/stringutil.h"
 
-#ifdef USE_RE2
+#ifdef I18N_PHONENUMBERS_USE_RE2
 #include "phonenumbers/regexp_adapter_re2.h"
 #else
 #include "phonenumbers/regexp_adapter_icu.h"
-#endif  // USE_RE2
+#endif  // I18N_PHONENUMBERS_USE_RE2
 
 namespace i18n {
 namespace phonenumbers {
@@ -60,13 +60,13 @@ struct RegExpTestContext {
 class RegExpAdapterTest : public testing::Test {
  protected:
   RegExpAdapterTest() {
-#ifdef USE_RE2
+#ifdef I18N_PHONENUMBERS_USE_RE2
     contexts_.push_back(
         new RegExpTestContext("RE2", new RE2RegExpFactory()));
 #else
     contexts_.push_back(
         new RegExpTestContext("ICU Regex", new ICURegExpFactory()));
-#endif  // USE_RE2
+#endif  // I18N_PHONENUMBERS_USE_RE2
   }
 
   ~RegExpAdapterTest() {

